src/sagemaker/processing.py

```diff
@@ -233,6 +234,12 @@ def _normalize_args(
             kms_key (str): The ARN of the KMS key that is used to encrypt the
                 user code file (default: None).
         """
+        if code and is_pipeline_variable(code):
```

Comment:
Could we do this check in the `arguments` method [on the steps](https://github.com/aws/sagemaker-python-sdk/blob/dev/src/sagemaker/workflow/steps.py#L557) instead of in the job classes? I'd like to keep pipeline code as centralized as possible instead of spreading it all over the codebase

Reply:
Also, does this apply to other steps like TrainingStep or only processing?

Reply:
> Could we do this check in the arguments method [on the steps](https://github.com/aws/sagemaker-python-sdk/blob/dev/src/sagemaker/workflow/steps.py#L557) instead of in the job classes?

We will have an improvement for Compatible Subclasses (implemented by Jerry), it will finally 1) remove the `code` parameter from the ProcessingStep and 2) let users call processing job `.run(code=...)` directly. Then 3) pass the outcome of the `.run()` to the ProcessingStep.
I was supposed to add the check this way to reduce code change for adopting the Compatible Subclasses Improvement but to make pipeline code centralized I can definitely move the check to ProcessingStep and add a TODO item there for possible changes relates to Compatible Subclasses Improvement